drivers/entropy/entropy_nrf5.c

```diff
@@ -100,3 +103,30 @@ DEVICE_AND_API_INIT(entropy_nrf5, CONFIG_ENTROPY_NAME,
 		    entropy_nrf5_init, &entropy_nrf5_data, NULL,
 		    PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEVICE,
 		    &entropy_nrf5_api_funcs);
+#else
+#include <bluetooth/crypto.h>
+
+static int entropy_nrf5_get_entropy(struct device *device, u8_t *buf, u16_t len)
+{
+	/* ATTENTION: calling this function implies that a bt init function has
+	 * been called, like bt_gatt_init(). If not, bt_rand will crash with a
```

Comment:
AFAIK we don't have a public bt_gatt_init() call, rather it's called indirectly through the public bt_enable() API. Also note that there are two bt_rand() implementations: one in subsys/bluetooth/host/crypto.c and another in subsys/bluetooth/controller/hci/hci.c. It's the latter that will always be chosen on nRF targets (which this patch seems to apply to) and I don't see anything in its implementation that'd directly depend on bt_enable(). It's possible however that the controller HCI driver's open() call does some initialization, in which case referring to bt_enable() is indeed the right thing.

Reply:
We have this call chain bt_rand() -> rand_get() -> get(). The first parameter of get is a ```static struct rand *rng_thr;``` which is set by the function rand_init();

If rand_init() is not called, the get() call will crash. We could also call rand_init() directly in the driver to prevent the crash, but I could be redondant if the app has already called ble_enable().

About the bt_gatt_init(), the function is public and directly callable in the app. I don't know if it's meant to be that way.

Reply:
What exactly makes you think that bt_gatt_init() is public? Public APIs are always available through include/bluetooth/ but bt_gatt_init() is available through subsys/bluetooth/host/gatt_internal.h.

Reply:
At first, I thought that this function was public as it can be called from the application (because gatt_internal.h is included in other header files).  

I have changed the comment to bt_enable()